### ISSUE TITLE
java.util.concurrent.CompletionException: java.lang.NullPointerException on ConfidentialClientApplication.acquireToken(ClientCredentialParameters)

### ISSUE THREAD

**Issue by rotj** [AzureAD/microsoft-authentication-library-for-java#253]
I create a sample application based on [ms-identity-java-daemon](https://github.com/Azure-Samples/ms-identity-java-daemon/blob/master/src/main/java/ClientCredentialGrant.java).

```
	private static IAuthenticationResult getAccessTokenByClientCredentialGrant() throws Exception {

		ConfidentialClientApplication app = ConfidentialClientApplication.builder(
				CONFIDENTIAL_CLIENT_ID,
				ClientCredentialFactory.createFromSecret(CONFIDENTIAL_CLIENT_SECRET))
				.authority(TENANT_SPECIFIC_AUTHORITY)
				.build();

		// With client credentials flows the scope is ALWAYS of the shape "resource/.default", as the
		// application permissions need to be set statically (in the portal), and then granted by a tenant administrator
		ClientCredentialParameters clientCredentialParam = ClientCredentialParameters.builder(
				Collections.singleton(GRAPH_DEFAULT_SCOPE))
				.build();

		CompletableFuture<IAuthenticationResult> future = app.acquireToken(clientCredentialParam);
		return future.get();
	}
```

Running as a desktop app, future gets a result of com.microsoft.aad.msal4j.AuthenticationResult with an accessToken.

When I copy the code to an HttpServlet class running on WebSphere, the result of future is `java.util.concurrent.CompletionException: java.lang.NullPointerException`.

```
[6/25/20 11:40:31:554 EDT] 000000c2 SystemErr     R java.util.concurrent.CompletionException: java.lang.NullPointerException
[6/25/20 11:40:31:555 EDT] 000000c2 SystemErr     R 	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:284)
[6/25/20 11:40:31:555 EDT] 000000c2 SystemErr     R 	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:291)
[6/25/20 11:40:31:556 EDT] 000000c2 SystemErr     R 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1617)
[6/25/20 11:40:31:556 EDT] 000000c2 SystemErr     R 	at java.lang.Thread.run(Thread.java:820)
[6/25/20 11:40:31:556 EDT] 000000c2 SystemErr     R Caused by: java.lang.NullPointerException
[6/25/20 11:40:31:557 EDT] 000000c2 SystemErr     R 	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.logException(AuthenticationResultSupplier.java:139)
[6/25/20 11:40:31:557 EDT] 000000c2 SystemErr     R 	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:88)
[6/25/20 11:40:31:559 EDT] 000000c2 SystemErr     R 	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:17)
[6/25/20 11:40:31:559 EDT] 000000c2 SystemErr     R 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1615)
[6/25/20 11:40:31:560 EDT] 000000c2 SystemErr     R 	... 1 more
```

Edit:
I did a more thorough trace and found the problem. Looks like I need to add a certificate to the server key store.
`com.microsoft.aad.msal4j.MsalClientException: javax.net.ssl.SSLHandshakeException: com.ibm.jsse2.util.h: PKIX path building failed: com.ibm.security.cert.IBMCertPathBuilderException: unable to find valid certification path to requested target`

I retrieved the certificate from graph.microsoft.com:443  and the error went away.

**Comment by theseyi** [comment]
What was your solution to this issue @rotj ? I'm getting a similar "javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target"

**Comment by rotj** [comment]
In Websphere Admin Console, I went to **Security > SSL certificate and key management > Key stores and certificates > NodeDefaultTrustStore > Signer certificates > Retrieve from port** and retrieved the certificate from graph.microsoft.com port 443. I also retrieved the certificate from login.microsoftonline.com to be safe.

If you use a different webserver, the process for adding a certificate will be different.
